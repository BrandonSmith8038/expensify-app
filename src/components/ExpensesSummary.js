import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpenseTotal from '../selectors/expenses-total'



export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'

    const formatedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')

    return (
        <div>
            <h1>Currently Viewing {expenseCount} {expenseWord} totaling {formatedExpensesTotal}</h1>
        </div>
    )
};

const mapStateToProps = state => {

    const visibileExpenses =  selectExpenses(state.expenses, state.filters)

    return{
        expenseCount: visibileExpenses.length,
        expensesTotal: selectExpenseTotal(visibileExpenses)

    }


};

export default connect(mapStateToProps)(ExpensesSummary);