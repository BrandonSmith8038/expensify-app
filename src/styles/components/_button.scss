.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-weight: 300;
  font-size: 30px;
  line-height: 1;
  padding: 20px;
  text-decoration: none;
}

.button--create {
  font-size: 18px;
  padding: 15px;
}

.button--link {
  background: none;
  font-size: 18px;
}

.button--remove {
  background: #888;
  font-size: 16px;
  padding: 15px;
  margin-bottom: 75px;
}
