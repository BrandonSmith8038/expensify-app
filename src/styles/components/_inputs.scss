.text-input {
    border: 1px solid #cacccd;
    font-size: 20px;
    font-weight: 300;
    height: 50px;
    padding: 12px;
}

.select {
    @extend .text-input;

}

.textarea {
    @extend .text-input;
    height: 100px;
}