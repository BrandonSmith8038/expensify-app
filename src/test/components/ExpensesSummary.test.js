import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import selectExpenseTotal from '../../selectors/expenses-total';
import expenses from '../fixtures/expenses';

test('Should render ExpensesSummary with singular expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={235} />
  );

  expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary with plural expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={2} expensesTotal={5325452345} />
  );

  expect(wrapper).toMatchSnapshot();
});
