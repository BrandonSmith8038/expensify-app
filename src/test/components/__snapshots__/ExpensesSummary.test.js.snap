// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary with plural expenses 1`] = `
<div>
  <h1>
    Currently Viewing 
    2
     
    expenses
     totaling 
    $53,254,523.45
  </h1>
</div>
`;

exports[`Should render ExpensesSummary with singular expense 1`] = `
<div>
  <h1>
    Currently Viewing 
    1
     
    expense
     totaling 
    $2.35
  </h1>
</div>
`;
